File: src/main/java/um/tesoreria/report/hexagonal/chequeras/domain/model/Lectivo.java
package um.tesoreria.report.hexagonal.chequeras.domain.model;

import lombok.*;
import java.time.OffsetDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Lectivo {
    private Integer lectivoId;
    private String nombre;
    private OffsetDateTime fechaInicio;
    private OffsetDateTime fechaFinal;
}

File: src/main/java/ar/edu/um/tesoreria/factura/rest/kotlin/model/Lectivo.kt
package ar.edu.um.tesoreria.factura.rest.kotlin.model

import ar.edu.um.tesoreria.factura.rest.model.Auditable
import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.OffsetDateTime

@Entity
@Table
data class Lectivo(

    @Id @Column(name = "lec_id")
    var lectivoId: Int? = null,

    @Column(name = "lec_nombre")

    var nombre: String = "",

    @Column(name = "lec_inicio")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaInicio: OffsetDateTime? = null,

    @Column(name = "lec_fin")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaFinal: OffsetDateTime? = null

) : Auditable()
